### ISSUE TITLE
Interactive splitting instantiates meta

### ISSUE THREAD

**Issue by GoogleCodeExporter** [agda/agda#1611]
```agda

open import Common.Prelude

data D : Bool → Set where
  dt : D true
  df : D false

Works : ∀{b} → D b → Set
Works dt = Bool
Works df = Bool

Fails : ∀{b} → D _ → Set
Fails dt = Bool
Fails df = Bool

-- false != true of type Bool
-- when checking that the pattern df has type D true

-- SHOULD BE:
-- Don't know whether to split on dt

-- Interactive splitting gives just one case.

F : ∀{b} → D _ → Set
F x = {!x!}

-- C-c C-c gives only case for dt
-- F dt = {!!}

```

Original issue reported on code.google.com by `andreas....@gmail.com` on 15 Jul 2015 at 2:48

**Comment by andreasabel** [comment]
Still open today (2017-07-28), and reported again, see #2621.

**Comment by andreasabel** [comment]
Fixed by @jespercockx along #2621.
